plugin/src/main/groovy/com/btkelly/gnag/api/GitHubApi.java

```diff
@@ -53,41 +57,78 @@ public GitHubApi(final GitHubExtension gitHubExtension) {
         Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl(baseUrl)
                 .client(okHttpClient)
+                .addConverterFactory(DiffParserConverterFactory.create())
                 .addConverterFactory(GsonConverterFactory.create())
                 .build();
 
         gitHubApiClient = retrofit.create(GitHubApiClient.class);
     }
 
-    public Status postGitHubComment(String comment) {
+    public void postGitHubPRCommentAsync(final String comment) {
+        gitHubApiClient.postPRComment(new GitHubPRComment(comment), gitHubExtension.getIssueNumber())
+                .enqueue(new DefaultCallback<>());
+    }
+
+    public void postUpdatedGitHubStatusAsync(final GitHubStatusType gitHubStatusType, final String prSha) {
+        gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), prSha)
+                .enqueue(new DefaultCallback<>());
+    }
 
+    @Nullable
+    public GitHubPRDetails getPRDetailsSync() {
         try {
-            Response<GitHubComment> gitHubCommentResponse = gitHubApiClient.postComment(new GitHubComment(comment), gitHubExtension.getIssueNumber()).execute();
-            return gitHubCommentResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            Response<GitHubPRDetails> gitHubPRDetailsResponse = gitHubApiClient.getPRDetails(gitHubExtension.getIssueNumber()).execute();
+            return gitHubPRDetailsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return null;
         }
     }
 
-    public Status postUpdatedGitHubStatus(GitHubStatusType gitHubStatusType, String sha) {
-
+    @NotNull
+    public List<Diff> getPRDiffsSync() {
         try {
-            Response<GitHubStatus> gitHubStatusResponse = gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), sha).execute();
-            return gitHubStatusResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            final Response<List<Diff>> gitHubPRDiffsResponse
+                    = gitHubApiClient.getPRDiffs(gitHubExtension.getIssueNumber()).execute();
+            
+            return gitHubPRDiffsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return new ArrayList<>();
         }
     }
 
-    public GitHubPullRequest getPullRequestDetails() {
+    public void postGitHubInlineCommentSync(
```

Comment:
Hmm not sure exactly why that would happen. We can revisit later as this seems like an enhancement, I'd like to get the line commenting in there.